crates/bitwarden-core/src/auth/login/mod.rs

```diff
@@ -1,94 +1,78 @@
 #[cfg(feature = "internal")]
-pub(crate) use password::login_password;
-#[cfg(feature = "internal")]
-pub use password::PasswordLoginRequest;
-#[cfg(any(feature = "internal", feature = "secrets"))]
-pub use password::PasswordLoginResponse;
+pub use password::*;
+
 #[cfg(feature = "internal")]
 mod two_factor;
 #[cfg(feature = "internal")]
-pub(crate) use two_factor::send_two_factor_email;
-#[cfg(feature = "internal")]
-pub use two_factor::{TwoFactorEmailRequest, TwoFactorProvider, TwoFactorRequest};
+pub use two_factor::*;
 
 #[cfg(feature = "internal")]
 mod api_key;
 #[cfg(feature = "internal")]
-pub(crate) use api_key::login_api_key;
-#[cfg(feature = "internal")]
-pub use api_key::{ApiKeyLoginRequest, ApiKeyLoginResponse};
+pub use api_key::*;
 
 #[cfg(feature = "internal")]
 mod auth_request;
 #[cfg(feature = "internal")]
-pub use auth_request::NewAuthRequestResponse;
-#[cfg(feature = "internal")]
-pub(crate) use auth_request::{complete_auth_request, send_new_auth_request};
+pub use auth_request::*;
 
 #[cfg(feature = "secrets")]
 mod access_token;
 #[cfg(feature = "secrets")]
-pub(super) use access_token::login_access_token;
-#[cfg(feature = "secrets")]
-pub use access_token::{AccessTokenLoginRequest, AccessTokenLoginResponse};
+pub use access_token::*;
 
-#[cfg(feature = "internal")]
-pub(crate) async fn request_prelogin(
-    client: &Client,
-    email: String,
-) -> Result<PreloginResponseModel> {
-    let request_model = PreloginRequestModel::new(email);
-    let config = client.internal.get_api_configurations().await;
-    Ok(accounts_prelogin_post(&config.identity, Some(request_model)).await?)
-}
+#[derive(Debug, thiserror::Error)]
+pub enum LoginError {
+    #[error(transparent)]
+    Api(#[from] crate::ApiError),
+    #[error(transparent)]
+    Crypto(#[from] bitwarden_crypto::CryptoError),
+    #[error(transparent)]
+    Serde(#[from] serde_json::Error),
+    #[error(transparent)]
+    InvalidBase64(#[from] base64::DecodeError),
 
-#[cfg(feature = "internal")]
-pub(crate) fn parse_prelogin(response: PreloginResponseModel) -> Result<Kdf> {
-    use std::num::NonZeroU32;
+    #[error(transparent)]
+    MissingField(#[from] crate::MissingFieldError),
+    #[error(transparent)]
+    JWTTokenParse(#[from] super::JWTTokenParseError),
+    #[error(transparent)]
+    Prelogin(#[from] PreloginError),
+    #[error(transparent)]
+    EncryptionSettings(#[from] crate::client::encryption_settings::EncryptionSettingsError),
+    #[error(transparent)]
+    AccessTokenInvalid(#[from] super::AccessTokenInvalidError),
+    #[error(transparent)]
+    NotAuthenticated(#[from] super::NotAuthenticatedError),
+    #[cfg(feature = "secrets")]
+    #[error(transparent)]
+    StateFile(#[from] crate::secrets_manager::state::StateFileError),
+    #[error("Error parsing Identity response: {0}")]
+    IdentityFail(crate::auth::api::response::IdentityTokenFailResponse),
+
+    #[error("Access token doesn't contain email")]
+    AccessTokenMissingEmail,
```

Comment:
It's a bit unfortunate that we have `AccessTokenInvalid` meaning the secrets manager access token, and `AccessTokenMissingEmail` meaning the JWT access token. Not related to this PR as we've had this naming before, but now they are part of the same enum it feels worse. Maybe we can rename this to `JWTTokenMissingEmail`?